### ISSUE TITLE
libPMacc using algorithm::math instead of ::math

### ISSUE THREAD

**Issue by PrometheusPi** [ComputationalRadiationPhysics/picongpu#1338]
While rewriting the particle to field interpolation for the pusher I found the following strange behavior.
The code is using

``` c++
{
using namespace PMacc;
...
}
```

thus `math::floor(...)` should reference to `PMacc::math::floor`. But the compiler searches for `floor` in `PMacc::algorithms::math`, where there is no `floor`.

Using  `PMacc::math::floor(...)` **solves** the problem. But why is the compiler looking in the namespace `algorithms`?

There is no `using namespace *algorithms` anywhere in libPMacc or PIConGPU. 
This might relate to #846.

**Comment by ax3l** [comment]
The problem is that there is a `math` namespace in both `PMacc` and `PMacc::algorithms`.

In what namespace scope do you use the `using namespace PMacc`?

**Comment by PrometheusPi** [comment]
@ax3l I use it in the namespace `picongpu` in [`InterpolationForPusher.hpp`](https://github.com/ComputationalRadiationPhysics/picongpu/blob/190af1c57a21545cce9b283604f89ba4c78a1997/src/picongpu/include/particles/InterpolationForPusher.hpp).

**Comment by ax3l** [comment]
greping did not lead to immediate results for me today, either.

ccing @erikzenker @psychocoderHPC

**Comment by erikzenker** [comment]
Do you want to use it like the following code shows ? : 

    namespace picongpu
    {
    using namespace PMacc;

    namespace particles
    {
    namespace interpolationMemoryPolicy
    {
    ...
    const T_PosType pos_floor = math::floor(pos);
    ...
    }
    }
    }

?

**Comment by PrometheusPi** [comment]
@erikzenker exactly!

**Comment by erikzenker** [comment]
Can't you put the `using namespace PMacc;` into the interpolationMemoryPolicy namespace ?

**Comment by PrometheusPi** [comment]
@erikzenker Yes I could. I used `PMacc::particles::interpolationMemoryPolicy` directly - that worked. But I am still confused, why we use the other `math` namespace.

**Comment by erikzenker** [comment]
I am confused as well. I tried to reproduce your error locally, but without success. I would suggest to put the `using namespace PMacc` statement as near as possible to the place where it is needed. But, otherwise maybe @psychocoderHPC has an idea.

**Comment by ax3l** [comment]
> I would suggest to put the using namespace PMacc statement as near as possible to the place where it is needed.

that is actually what one should do, otherwise the higher namespaces will be polluted for all files coming after you since we are header-only.

**Comment by PrometheusPi** [comment]
Putting `using namespace PMacc;` in `namespace interpolationMemoryPolicy` does not work either. 
Same error as before: 
```
.../src/picongpu/include/particles/interpolationMemoryPolicy/ShiftToValidRange.hpp(43): error: no instance of function template "PMacc::algorithms::math::floor" matches the argument list
```

**Comment by ax3l** [comment]
really weird what causes this. as a clean implementation you can for now do
```C++
namespace picongpu
{
namespace particles
{
namespace interpolationMemoryPolicy
{
    namespace pmath = PMacc::math; // or ::PMacc::math if necessary
    ...
    const T_PosType pos_floor = pmath::floor(pos);
    ...
}
}
}
```

**Comment by PrometheusPi** [comment]
Yes, see [here](https://github.com/ComputationalRadiationPhysics/picongpu/blob/7972216a23b08066df8d000f2cee2237be8fd808/src/picongpu/include/particles/interpolationMemoryPolicy/ShiftToValidRange.hpp#L41)

**Comment by PrometheusPi** [comment]
I found the reason for this reference to `algorithms::math`: It is a [namespace renaming in `/picongpu/include/simulation_types.hpp`](https://github.com/ComputationalRadiationPhysics/picongpu/blob/5f82fc19bb78ef0bb6ebe42db44a628454be5c94/src/picongpu/include/simulation_types.hpp#L68).

**Comment by erikzenker** [comment]
y? ...time to git blame

**Comment by ax3l** [comment]
ah yes, let us rename this to, e.g., `pmath` or let's remove it completely

**Comment by psychocoderHPC** [comment]
I found the problem: In [Vector.hpp](https://github.com/ComputationalRadiationPhysics/picongpu/blob/dev/src/libPMacc/include/math/vector/Vector.hpp#L765-L772) is a definition of floor. This floor is not the floor from `algorithms/math` and used hard coded float.
The correct `algorithms::math::floor` is currently not defined for `PMacc::math::Vector` and therefore the compiler triggered the error.

**Comment by psychocoderHPC** [comment]
should b solved with #1472
